on-premise-docker-registry/README.md

```diff
@@ -0,0 +1,191 @@
+- Customers who do not wish to set up contract with public cloud vendor.
+- other reasons...
+
+These customers can consider leveraging a Docker registry running on-premise. Docker images are pulled then by connecting the Kubernetes cluster to the registry using Connectivity Proxy and Cloud Connector.
+
+See the following sample flow of how this can be achieved
+
+![flow](assets/on-prem-docker-reg.png)
+
+Here, nginx as a reverse proxy is used to forward the HTTP requests for pulling Docker images from the on-premise Docker registry using Connectivity Proxy and Cloud Connector.
+
+>**NOTE:** In this sample, nginx as a reverse proxy is used to pull the images from an on-premise Docker registry. The setup relies on the Docker registry API v2 and proves the concept. This can be replaced with another reverse proxy or a custom implementation based on API and requirements.
+
+## Prerequisites
+
+- [SAP BTP, Kyma runtime instance](../prerequisites/#kyma)
+- [Docker](../prerequisites/#docker)
+- [GNU Make](https://www.gnu.org/software/make/)
+- [Kubernetes tooling](../prerequisites/#kubernetes)
+- [Cloud Connector on your machine or test system](../prerequisites/#sap-cloud-connector)
+- [Connectivity Proxy instance configured in Kyma](https://help.sap.com/docs/btp/sap-business-technology-platform/configure-sap-btp-connectivity-in-kyma-environment)
+- [OpenSSL](https://www.openssl.org/) or another similar tool to generate the certificates
+- [htpasswd](https://httpd.apache.org/docs/2.4/programs/htpasswd.html) installed
+
+## Setup
+
+The commands have been verified on OSX. However, it should be possible to adapt them for a windows laptop.
+
+### Environment variables
+
+- Export the following environment variables:
+
+```shell
+export KUBECONFIG=<path-to-kubeconfig>
+export NAMESPACE={kyma-namespace-used-for-this-sample}
+export CLUSTER_DOMAIN={kyma-cluster-domain}
+export REG_USER_NAME={docker-registry-user}
+export REG_USER_PASSWD={docker-registry-password}
+export EMAIL={your-email}
+```
+
+### On-premise Docker registry
+
+In this sample, we set up a simple Docker registry running on your machine. This can be replaced with a productive alternate deployed in your corporate network. The only requirement is that it should be reachable using Cloud Connector.
+
+- Generate the self-signed certificate for enabling HTTPS for the Docker registry. Be sure to specify the `CN` as `myregistry.kyma`
+
+```shell
+make generate-self-signed-cert
+```
+
+- Add the newly created certificate to your trust storage
+
+```shell
+make trust-self-signed-cert
+```
+
+- Generate the `htpasswd`. This is used for accessing the on-premise Docker registry.
+
+```shell
+make generate-htpasswd
+```
+
+- Start the Docker registry server
+
+```shell
+make start-docker-registry
+```
+
+- Add DNS entry to the `/etc/hosts` file
+
+```shell
+127.0.0.1 myregistry.kyma
+```
+
+- Configure Cloud Connector to access the on-premise Docker registry.
+  ![cc-config](assets/cc-config.png)
```

Comment:
If we expect that users will want to copy and paste the values, maybe a list in markdown would be better than a screenshot.

Reply:
```suggestion
   - Back-end type: Non-SAP System
   - Protocol: HTTPS
   - Internal Host: `myregistry.kyma`
   - Internal Port: `25002`
   - Principal Type: None
   - Host In Request Header: Use Internal Host
```
If any of those values are default and don't need to be touched; we could consider removing them from the list (so that only the ones that must be adjusted remain) and adding the statement "Keep the default for all other values."

Reply:
none are default